### PR TITLE
Add possibility to define nested structures in DisksConfig

### PR DESCRIPTION
**Pull request by giggsoff**
To use stripe of two pairs of mirrored disks we should define
DisksConfig without disks with array_type DISKS_ARRAY_TYPE_RAID0 with
two children with properly defined disks inside and with array_type
DISKS_ARRAY_TYPE_RAID1 and empty children

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: api/proto/config/storage.proto [hunk 1]
```diff
@@ -207,20 +207,38 @@ enum DiskConfigType {
...
-//If we want to replace disk we should fill old_disk to be replaced with disk
-//Progress of operation is expected to be available in info messages
+//All disks defined in disks field expected to have array type defined in array_type
+//To support nested topologies we can use children field
+//
+//For example to use stripe of two pairs of mirrored disks we should define
+//DisksConfig without disks with array_type DISKS_ARRAY_TYPE_RAID0
+//with two children with properly defined disks inside and with array_type DISKS_ARRAY_TYPE_RAID1
+//and empty children
 message DisksConfig {
```
Comment (vk-en) [inline api/proto/config/storage.proto:240]: Maybe should rename it to PoolConfig? if here the context is only for ZFS

**Review by eriknordmark — APPROVED** [review]
LGTM

But don't we also need the same API structure for the info messages?

**Review by zedi-pramodh — APPROVED** [review]
LGTM

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**